server/src/main/java/org/elasticsearch/cluster/service/ClusterService.java

```diff
@@ -263,4 +264,7 @@ public <T extends ClusterStateTaskListener> void submitStateUpdateTask(
         masterService.submitStateUpdateTask(source, task, config, executor);
     }
 
+    public void setTaskManager(TaskManager taskManager) {
+        masterService.setTaskManager(taskManager);
```

Comment:
On the surface it looks possible to construct the `TaskManager` outside the `TransportService` and thus be able to pass it in properly to the constructor of the `MasterService`. I'd prefer that if possible.

Reply:
I've [tried this refactoring](https://github.com/elastic/elasticsearch/pull/87917/files/c1c7c4827412e867b6f9603ea1c4a0f7bfe8d83b..6602e7375a8553ac387070a01698e61da409a1b2), see what you think. The impact was greater that I would have liked, because the `TaskManager` instance was being fetched from the `TransportService` instance. I had to rework things so that I could pass in the same `TaskManager` to the `ClusterService` and `TransportService`.

Reply:
That part looks great, thanks for undertaking that.